cartography/graph/sanitizer.py

```diff
@@ -0,0 +1,293 @@
+"""
+Data sanitization and formatting utilities for Cartography node schemas.
+
+This module provides functionality to clean up and format data dictionaries based on
+CartographyNodeSchema definitions. It ensures that data conforms to the expected
+structure and types defined in the schema, while automatically formatting values
+according to their PropertyRef specifications.
+
+The main functionality includes:
+- Filtering data to include only schema-defined properties
+- Automatic type conversion and formatting based on PropertyRef settings
+- Recursive handling of nested data structures
+- Error-resilient data processing with fallback mechanisms
+"""
+
+import logging
+from dataclasses import asdict
+from datetime import datetime
+from typing import Any
+
+from dateutil.parser import parse as parse_date
+
+from cartography.models.core.common import PropertyRef
+from cartography.models.core.nodes import CartographyNodeSchema
+from cartography.models.core.relationships import CartographyRelSchema
+
+logger = logging.getLogger(__name__)
+
+
+def data_dict_cleanup(
```

Comment:
cleanup is used in the project to describe keeping things fresh

this should probably be 'format' related or normalization related